Dockerfile

```diff
@@ -14,79 +14,125 @@
 # limitations under the License.
 #
 
-FROM --platform=linux/riscv64 riscv64/ubuntu:22.04 as sdk
-ARG LINUX_SOURCES_VERSION=5.15.63-ctsi-2
-ARG LINUX_SOURCES_FILEPATH=dep/linux-sources-${LINUX_SOURCES_VERSION}.tar.gz
-ARG RNDADDENTROPY_VERSION=3.0.0
-ARG RNDADDENTROPY_FILEPATH=dep/twuewand-$(RNDADDENTROPY_VERSION).tar.gz
-ARG BUILD_BASE=/opt/cartesi/
-
-# apt
+FROM ubuntu:22.04 as tools-env
+ARG IMAGE_KERNEL_VERSION=v0.19.1
+ARG LINUX_VERSION=6.5.9-ctsi-1
+ARG LINUX_HEADERS_URLPATH=https://github.com/cartesi/image-kernel/releases/download/${IMAGE_KERNEL_VERSION}/linux-libc-dev-riscv64-cross-${LINUX_VERSION}-${IMAGE_KERNEL_VERSION}.deb
+ARG BUILD_BASE=/opt/cartesi
+
+# Install dependencies
 # ------------------------------------------------------------------------------
-RUN DEBIAN_FRONTEND=noninteractive apt update && \
-    apt upgrade -y && \
-    apt install -y --no-install-recommends \
+ENV LINUX_HEADERS_FILEPATH=/tmp/linux-libc-dev-riscv64-cross-${LINUX_VERSION}-${IMAGE_KERNEL_VERSION}.deb
+
+RUN DEBIAN_FRONTEND=noninteractive apt-get update && \
+    apt-get upgrade -y && \
+    apt-get install -y --no-install-recommends \
       build-essential \
       ca-certificates \
       git \
-      protobuf-compiler \
-      rsync \
-      rust-all=1.58.1+dfsg1~ubuntu1-0ubuntu2 \
+      wget \
+      crossbuild-essential-riscv64 \
+      gcc-12-riscv64-linux-gnu \
+      g++-12-riscv64-linux-gnu \
       && \
-    rm -rf /var/lib/apt/lists/*
+    wget -O ${LINUX_HEADERS_FILEPATH} ${LINUX_HEADERS_URLPATH} && \
```

Comment:
As a simplification:
We could use Dockers `ADD` command to fetch this file instead, move the `sha256sum` at the top of this `RUN` and apt-get install everything at once.